### PR TITLE
Add support for .yml files located in work dir or in classpath

### LINKED ISSUE #1836: Jet should support "hazelcast-jet.yml" config file
See https://github.com/hazelcast/hazelcast/issues/16205

Both ".yaml" and ".yml" suffix should be accepted for the "hazelcast-jet" YAML config file.

### PR DESCRIPTION
**Pull request by eminn**
Checklist
- [x] Tags Set
- [x] Milestone Set
- [N/A] Any breaking changes are documented
- [N/A] New public APIs have `@Nonnull/@Nullable` annotations
- [N/A] New public APIs have `@since` tags in Javadoc
- [N/A] For code samples, code sample main readme is updated

Links to issues fixed (if any):

Fixes #1836

### CODE REVIEW

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderConfigResolutionTest.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+
+import static com.hazelcast.internal.config.DeclarativeConfigUtil.YAML_ACCEPTED_SUFFIXES_STRING;
+import static com.hazelcast.jet.impl.config.JetDeclarativeConfigUtil.SYSPROP_JET_CONFIG;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
```
Comment (jbartok) [inline hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderConfigResolutionTest.java]: Why serial runner, is there a reason why these tests can't be run in parallel?

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderConfigResolutionTest.java [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+public class YamlJetConfigBuilderConfigResolutionTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    private JetDeclarativeConfigFileHelper helper = new JetDeclarativeConfigFileHelper();
+
+    @Before
```
Comment (jbartok) [inline hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderConfigResolutionTest.java]: Bit weird for the `tearDown()` method to be annotated with `@Before`... Mistake?

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderConfigResolutionTest.java [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+import org.junit.runner.RunWith;
+
+import java.io.File;
+
+import static com.hazelcast.internal.config.DeclarativeConfigUtil.YAML_ACCEPTED_SUFFIXES_STRING;
+import static com.hazelcast.jet.impl.config.JetDeclarativeConfigUtil.SYSPROP_JET_CONFIG;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
+public class YamlJetConfigBuilderConfigResolutionTest {
```
Comment (jbartok) [inline hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderConfigResolutionTest.java]: Name is quite a mouthful, maybe something shorter?

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderMemberConfigResolutionTest.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+
+import static com.hazelcast.internal.config.DeclarativeConfigUtil.SYSPROP_MEMBER_CONFIG;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
```
Comment (jbartok) [inline hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderMemberConfigResolutionTest.java]: Same question, can't you run them in parallel?

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderMemberConfigResolutionTest.java [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+public class YamlJetConfigBuilderMemberConfigResolutionTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    private JetDeclarativeConfigFileHelper helper = new JetDeclarativeConfigFileHelper();
+
+    @Before
```
Comment (jbartok) [inline hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderMemberConfigResolutionTest.java]: Same as before, intentional?

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderConfigResolutionTest.java (hunk 1 above)
Comment (eminn) [reply hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderConfigResolutionTest.java]: yes, they are creating some files on runtime with the same name so they should be run serially

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderConfigResolutionTest.java (hunk 2 above)
Comment (eminn) [reply hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderConfigResolutionTest.java]: it should clear out the system property if present, so we also run this method before each run.

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderConfigResolutionTest.java (hunk 3 above)
Comment (eminn) [reply hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderConfigResolutionTest.java]: any suggestions ?

Comment (jbartok) [reply hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderConfigResolutionTest.java]: Maybe at least leave out the second `Config`, like `YamlJetConfigBuilderResolutionTest`? Or even `YamlJetConfigResolutionTest`? Not sure.

**Pull request merged**